### ISSUE TITLE
Server fails when uploading MaxMind DB on a live instance

### ISSUE THREAD

**Issue by yakkomajuri** [PostHog/plugin-server#33]
Was setting up the MaxMind plugin today on a live instance. Uploading the DB file led to this:

<img width="600" alt="Screenshot 2020-12-02 at 15 13 00" src="https://user-images.githubusercontent.com/38760734/100891581-5089dc80-34b1-11eb-9d2f-b372d32260dd.png">

Heroku Logs:

```shell
2020-12-02T15:11:10.466690+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/celery/backends/redis.py", line 177, in _consume_from
2020-12-02T15:11:10.466691+00:00 app[worker.1]: self._pubsub.subscribe(key)
2020-12-02T15:11:10.466691+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/contextlib.py", line 131, in __exit__
2020-12-02T15:11:10.466691+00:00 app[worker.1]: self.gen.throw(type, value, traceback)
2020-12-02T15:11:10.466699+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/celery/backends/redis.py", line 130, in reconnect_on_error
2020-12-02T15:11:10.466699+00:00 app[worker.1]: self._ensure(self._reconnect_pubsub, ())
2020-12-02T15:11:10.466700+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/celery/backends/redis.py", line 355, in ensure
2020-12-02T15:11:10.466700+00:00 app[worker.1]: return retry_over_time(
2020-12-02T15:11:10.466700+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/kombu/utils/functional.py", line 344, in retry_over_time
2020-12-02T15:11:10.466701+00:00 app[worker.1]: return fun(*args, **kwargs)
2020-12-02T15:11:10.466701+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/celery/backends/redis.py", line 115, in _reconnect_pubsub
2020-12-02T15:11:10.466701+00:00 app[worker.1]: metas = self.backend.client.mget(self.subscribed_to)
2020-12-02T15:11:10.466702+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/redis/client.py", line 1644, in mget
2020-12-02T15:11:10.466702+00:00 app[worker.1]: return self.execute_command('MGET', *args, **options)
2020-12-02T15:11:10.466703+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/redis/client.py", line 875, in execute_command
2020-12-02T15:11:10.466703+00:00 app[worker.1]: conn = self.connection or pool.get_connection(command_name, **options)
2020-12-02T15:11:10.466703+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/redis/connection.py", line 1185, in get_connection
2020-12-02T15:11:10.466704+00:00 app[worker.1]: connection.connect()
2020-12-02T15:11:10.466704+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/redis/connection.py", line 561, in connect
2020-12-02T15:11:10.466705+00:00 app[worker.1]: self.on_connect()
2020-12-02T15:11:10.466705+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/redis/connection.py", line 637, in on_connect
2020-12-02T15:11:10.466705+00:00 app[worker.1]: auth_response = self.read_response()
2020-12-02T15:11:10.466706+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/redis/connection.py", line 734, in read_response
2020-12-02T15:11:10.466706+00:00 app[worker.1]: response = self._parser.read_response()
2020-12-02T15:11:10.466706+00:00 app[worker.1]: File "/app/.heroku/python/lib/python3.8/site-packages/redis/connection.py", line 333, in read_response
2020-12-02T15:11:10.466707+00:00 app[worker.1]: raise error
2020-12-02T15:11:10.466707+00:00 app[worker.1]: redis.exceptions.ConnectionError: max number of clients reached
```

**Comment by mariusandra** [comment]
Thanks for testing!

`redis.exceptions.ConnectionError: max number of clients reached` :(. 

I'm assuming this is Heroku with the free Redis addon that has a 20 connection limit?

The unreleased code in `master` [halves](https://github.com/PostHog/posthog-plugin-server/pull/28) the number of redis clients needed from 4 to 2, so it *could* have already fixed this. Still some work to do before we can cut the next release. Next week latest.

**Comment by mariusandra** [comment]
Hey! Once https://github.com/PostHog/posthog/pull/2703 lands, this might be fixed. Sort of. Maybe. Not quite, but fixed enough?

I've been playing with the latest review app and ran into a different issue. While all worked fine with Redis, etc, I couldn't upload the 70MB file as I ran into a [30sec heroku router timeout](https://devcenter.heroku.com/articles/request-timeout). I was able to fully upload the smaller, 4MB GeoLite2-Country database `.mmdb` file and it worker well! All events had geodata. Just countries, but this might be enough for most people.

The real solution for the 30sec/70MB issue is to update the plugin so that it could also directly accept the `.tar.gz` files you download from maxmind. Then the plugin would extract the archive and get the data it needs.

**Comment by AxelNilsson** [comment]
I'm also having this issue but I'm running on Kubernetes with a nginx ingress. If you look at the response from the server I got a 413 error meaning the nginx blocked the request since the payload was too large. When I added `nginx.ingress.kubernetes.io/proxy-body-size: 100m` to my ingress config everything worked as expected. Technically that means this isn't a PostHog issue but I would assume a lot of users will have this issue. Just remember to reduce the nginx `body-size` to something much smaller, like 4m, after the upload has been completed.
